src/onnx/parse_resize.cpp

```diff
@@ -206,171 +297,202 @@ struct parse_resize : op_parser<parse_resize>
+            std::size_t out_elements = out_s.elements();
+            auto idx_op              = get_original_idx_op(coord_trans_mode);
+
+            // reshape input to one-dimension
+            std::vector<int64_t> rsp_lens = {static_cast<int64_t>(in_s.elements())};
+            args[0]                       = info.make_contiguous(args[0]);
+            auto rsp = info.add_instruction(make_op("reshape", {{"dims", rsp_lens}}), args[0]);
+
+            if(mode == "nearest")
+            {
+                std::vector<int> ind(out_elements);
 
-            // get the number of dimensions
-            std::size_t n_dim = out_lens.size();
-            std::vector<std::vector<std::size_t>> vv_ind(2, std::vector<std::size_t>(out_elements));
-            std::vector<std::vector<std::vector<std::size_t>>> vvv_ind(n_dim, vv_ind);
-            std::vector<std::vector<float>> delta(n_dim, std::vector<float>(out_elements));
+                // map out_idx to in_idx
+                auto nearest_op = get_nearest_op(nearest_mode);
+                shape_for_each(out_s, [&](const auto& out_idx_v, size_t out_idx) {
+                    std::vector<size_t> in_idx(out_idx_v.size());
+                    for(auto ii = 0; ii < in_lens.size(); ++ii)
+                    {
+                        auto idx_val =
+                            idx_op(in_lens[ii], out_lens[ii], out_idx_v[ii], vec_scale[ii]);
+                        in_idx[ii] = nearest_op(in_lens[ii], idx_val);
+                    }
 
-            shape_for_each(out_s, [&](const auto& out_idx_v, size_t out_idx) {
-                for(auto ii = 0; ii < in_lens.size(); ++ii)
-                {
-                    auto idx_val = idx_op(in_lens[ii], out_lens[ii], out_idx_v[ii], vec_scale[ii]);
-                    vvv_ind[ii][0][out_idx] = nearest_floor(in_lens[ii], idx_val);
-                    vvv_ind[ii][1][out_idx] = nearest_ceil(in_lens[ii], idx_val);
-                    delta[ii][out_idx]      = idx_val - vvv_ind[ii][0][out_idx];
-                }
-            });
+                    ind[out_idx] = static_cast<int64_t>(in_s.index(in_idx));
+                });
 
-            auto ind = calc_neighbor_points(
-                vvv_ind, 0, std::vector<std::vector<std::size_t>>(out_elements), in_s);
-            auto ind_lens = out_lens;
-            ind_lens[0] *= (std::size_t{1} << n_dim);
-            shape ind_s{shape::int32_type, ind_lens};
-            auto ins_ind = info.add_literal(literal(ind_s, ind));
-            auto data    = info.add_instruction(make_op("gather", {{"axis", 0}}), rsp, ins_ind);
-
-            auto dim_lens = out_lens;
-            dim_lens[0] *= (std::size_t{1} << (n_dim - 1));
-            for(std::size_t i = 0; i < n_dim; ++i)
+                shape ind_s{shape::int32_type, out_lens};
+                auto ins_ind = info.add_literal(literal(ind_s, ind));
+                return info.add_instruction(make_op("gather", {{"axis", 0}}), rsp, ins_ind);
```

Comment:
This version makes multiple gathers that progressively resize the tensor. This works because of the locality of the nearest algorithm and the regularity of the tensor indices. There is a way to use a single gather to do the same thing. @pfultz2 is making it one gather rather than several preferable or vice versa?

Reply:
If we do a single gather, the index array must be much larger--for instance, in resizing to 1000x1000 a 2-D index has 1,000,000 entries while 2 1-D indices have only 2,000.  I don't know if this impacts performance in a significant way.